stl/inc/algorithm

```diff
@@ -394,6 +793,194 @@ _NODISCARD pair<_FwdIt1, _FwdIt2> mismatch(
 }
 #endif // _HAS_CXX17
 
+#ifdef __cpp_lib_concepts
+namespace ranges {
+    // ALIAS TEMPLATE mismatch_result
+    template <class _In, class _Out>
+    using mismatch_result = in_in_result<_In, _Out>;
+
+    // VARIABLE ranges::mismatch
+    class _Mismatch_fn : private _Not_quite_object {

```

Comment:
Should we be considering powering the older algorithms with these to avoid repeated definitions?

Reply:
Yes, in the future ;)

Reply:
Future-Casey is going to love Past-Casey